### ISSUE TITLE
Console output is not shown on Release build

### ISSUE THREAD

**Issue by saki7** [godotengine/godot#56884]
### Godot version

master (846c14eee9bad75f31df94f7072331e85d24cff5)

### System information

Windows 10, GTX1080Ti (511.23), Vulkan

### Issue description

Since #56012 was merged, godot is eliminating some outputs which are directly written to standard outputs.
This behavior affects various 3rd-party libraries, such as Google Test, which writes output via `printf`. (Those are hard-coded 
and can't be configured from the application side)

The output vanishes only on `Release` build. I can see outputs on `Debug` build.
- godot: `target=release_debug`
- godot-cpp: `target=debug` and `target=release`, accordingly
- My app: GDExtension, building with either `Debug` (working fine) or **`Release` (bugged)**

I confirmed that reverting #56012 fixes this issue. I suppose some treatment is required for Windows platform.

Workarounds such as launching godot via cmd.exe (https://github.com/godotengine/godot/pull/56012#issuecomment-996575966) is not working.

### Steps to reproduce

```cpp
std::cout << "std::cout" << std::endl; // prints nothing
std::cerr << "std::cerr" << std::endl; // prints nothing
std::printf("std::printf"); // prints nothing
```

### Minimal reproduction project

n/a

**Comment by Calinou** [comment]
cc @bruvzg

**Comment by bruvzg** [comment]
You can build Godot with enabled console by adding `windows_subsystem=console` to the build command (including release export templates).

Seems like each library need to call following code on its own to redirect output, not sure if it's possible to solve it from the Godot side.

https://github.com/godotengine/godot/blob/8ae86f608a797d82a3546939181c830fa140e52b/platform/windows/os_windows.cpp#L88-L94

**Comment by saki7** [comment]
I am actually happy with the design decision that the default GUI is no longer showing the console (which pretty much makes sense). 

Do you have any clue why it is only bugged for Release build?

**Comment by bruvzg** [comment]
> Do you have any clue why it is only bugged for Release build?

I do not know the exact reason, but it's likely related to the different versions of CRT used by the editor and library.

On Windows, there's no one-to-one mapping between C file handles and OS file handles, library probably have its own set of C handles which is not affected by `freopen_s` calls in the engine.

**Comment by clayjohn** [comment]
@bruvzg Can you clarify the intended behaviour for me?

I build with ``windows_subsystem=console`` and my expectation is that output will be written to the console even from a child process. I.e. If I open the project manager and run a project (or edit a project) the console will remain open and capture the output from the editor, or running project. 

However, what I see now is that output is captured from the project manager and then the console is closed upon running or editing a project. The behaviour is the same if I run Godot from the command line. I.e. When running from the command line the process is ended as soon as I select "edit" or "run" and output is no longer captured.

**Comment by akien-mga** [comment]
Fixed by #58455.
